python/cudf/cudf/core/column/column.py

```diff
@@ -126,6 +125,46 @@ def _can_values_be_equal(left: DtypeObj, right: DtypeObj) -> bool:
     return False
 
 
+class MaskCAIWrapper:
+    # A wrapper that exposes the __cuda_array_interface__ of a mask that accounts for
+    # the mask being a bitmask in the mask size calculation.
+
+    def __init__(self, mask: Any) -> None:
+        self._mask = mask
+
+    @property
+    def __cuda_array_interface__(self) -> Mapping:
+        cai = self._mask.__cuda_array_interface__.copy()
+        cai["shape"] = (
+            plc.null_mask.bitmask_allocation_size_bytes(cai["shape"][0]),
+        )
+        return cai
+
+
+class ROCAIWrapper:
+    # A wrapper that exposes the __cuda_array_interface__ of a buffer as read-only to
+    # avoid copy-on-write issues.
+    def __init__(self, buffer: Buffer, mode: Literal["read", "write"]) -> None:
```

Comment:
Minor: I was a little confused with the read-only naming of this object conflating with the ability to pass `mode="write"`. IIUC these are different read/write semantics but just noting if you have ideas to distinguish the `RO` in `ROCAIWrapper`

Reply:
Actually I think this name is flat-out inaccurate at this point. When I named it I was working through a sequence of issues and I thought I was going to need a read-only component in addition to the lazy evaluation of the pointer, but ultimately that proved wrong. As with your other comment, I don't know if this code is going to stick around in its current form for very long, so I am going to merge as-is to keep things moving, but if I don't wind up simplifying the code enough to eliminate this class altogether I will rename it accordingly.